pkg/operator/diff/internal/diff.go

```diff
@@ -186,14 +189,58 @@ func (g *DiffGenerator) Run(oldModel, newModel model.Model) (model.Model, error)
 		var outputHasDefault bool
 		outputPkg.DefaultChannel, outputHasDefault = outputPkg.Channels[newPkg.DefaultChannel.Name]
 		if !outputHasDefault {
-			// Create a name-only channel since oldModel contains the channel already.
-			outputPkg.DefaultChannel = copyChannelNoBundles(newPkg.DefaultChannel, outputPkg)
+			// Set the defaultChannel using the priority of a channel when the default got filtered out
+			// If no channels with the Priority property, raise an error
+			if err := setDefaultChannel(outputPkg); err != nil {
+				return outputModel, err
+			}
 		}
 	}
 
 	return outputModel, nil
 }
 
+type ChannelPriorityPropList []property.Channel
+
+// setDefaultChannel sets the new default channel of a package if the old default channel got filtered out.
+// Throws an error if there are no channels with the Priority property
+func setDefaultChannel(outputPkg *model.Package) error {
+	p := make(ChannelPriorityPropList, len(outputPkg.Channels))
+	i := 0
+	for _, channel := range outputPkg.Channels {
+		var channelPriority property.Channel
+		for _, prop := range channel.Properties {
+			if prop.Type == "olm.channel" {
```

Comment:
It may be better to import the constant from `operator-registry`. Just in case the value changes for this property type https://github.com/operator-framework/operator-registry/blob/ce3bb44f7618726474a78f95852e30a2bfb0346a/alpha/property/property.go#L141.